subprojects/plugin-use/src/main/java/org/gradle/plugin/management/internal/autoapply/DefaultAutoAppliedPluginRegistry.java

```diff
@@ -72,9 +72,13 @@ private boolean shouldApplyGradleEnterprisePlugin(Settings settings) {
     }
 
     private static PluginRequestInternal createGradleEnterprisePluginRequest() {
+        // We use a version range here so that a user-provided version will be preferred (as long as it falls in the range).
+        // Without this, adding `--scan` to the command line can result in a different GE plugin being used.
+        String gePluginVersionRange = "(," + AutoAppliedGradleEnterprisePlugin.VERSION + "]";
```

Comment:
I'm not sure how much work it would be to change this around, but would it make more sense for this to be a rich version constraint, @ljacomet?

Reply:
AFAIK we do not have ways to inject constraints into the plugins block level resolution.